backbone/src/test/java/org/researchstack/backbone/ui/task/TaskViewModelTest.kt

```diff
@@ -0,0 +1,419 @@
+        assertEquals(taskViewModel.task, task)
+    }
+
+    @Test
+    fun testEdit_editShouldBeTrue_EditStepShouldBeUpdated_CancelMenuShouldBeTrueToHideCancelInTitleBar() {
+        // Assemble
+        val selectedEditStep = createCurrentStep("selectedEditStep")
+        taskViewModel = createViewModel()
+
+        // Act
+        taskViewModel.edit(selectedEditStep)
+
+        // Assert
+        Assert.assertEquals(taskViewModel.currentStep, selectedEditStep)
+        Assert.assertEquals(taskViewModel.editStep.value, selectedEditStep)
+        Assert.assertEquals(taskViewModel.editing, true)
+        Assert.assertEquals(taskViewModel.hideMenuItemCancel.value, true)
+    }
+
+
+    @Test
+    fun testRemoveUpdatedLayout_editShouldBeFalse_EditStepShouldBeUpdated_CancelMenuShouldBeFalseToShowCancelInTitleBar() {
+        // Assemble
+        taskViewModel = spy(createViewModel())
+        val currentStepMocked = createCurrentStep()
+        val nextStepMocked = createCurrentStep("nextStepMocked")
+        taskViewModel.currentStep = currentStepMocked
+        `when`(task!!.getStepAfterStep(any(), any())).thenReturn(nextStepMocked)
+        `when`(taskViewModel.isReviewStep(nextStepMocked)).thenReturn(true)
+
+        // Act
+        taskViewModel.removeUpdatedLayout()
+
+        // Assert
+        Assert.assertEquals(taskViewModel.editing, false)
+        Assert.assertEquals(taskViewModel.updateCancelEditInLayout.value, true)
+        Assert.assertEquals(taskViewModel.hideMenuItemCancel.value, false)
+    }
+
+    @Test
+    fun testNext_ShouldUpdateCurrentStepToNextStep_shouldUpdateCurrentStepEventToNextStep() {
+        // Assemble
+        taskViewModel = createViewModel()
+        val currentStepMocked = createCurrentStep()
+        val nextStepMocked = createCurrentStep("nextStepMocked")
+        taskViewModel.currentStep = currentStepMocked
+        `when`(task!!.getStepAfterStep(any(), any())).thenReturn(nextStepMocked)
+
+        // Act
+        taskViewModel.nextStep()
+
+        // Assert
+        Assert.assertEquals(taskViewModel.currentStep, nextStepMocked)
+        Assert.assertEquals(taskViewModel.currentStepEvent.value!!.step, nextStepMocked)
+    }
+
+    @Test
+    fun testPrevious_ShouldUpdateCurrentStepToPreviousStep_shouldUpdateCurrentStepEventToPreviousStep() {
+        // Assemble
+        taskViewModel = createViewModel()
+
+        val currentStepMocked = createCurrentStep()
+        val previousStepMocked = createCurrentStep("previousStepMocked")
+        taskViewModel.currentStep = currentStepMocked
+        `when`(task!!.getStepBeforeStep(any(), any())).thenReturn(previousStepMocked)
+
+        // Act
+        taskViewModel.previousStep()
+
+        // Assert
+        Assert.assertEquals(taskViewModel.currentStep, previousStepMocked)
+        Assert.assertEquals(taskViewModel.currentStepEvent.value!!.step, previousStepMocked)
+    }
+
+    @Test
+    fun checkPrevious_shouldCloseWhenPreviousIsNull() {
```

Comment:
`shouldClose` what? Can you better describe what this does?

Reply:
updated